UnityProject/Assets/Scripts/Items/Implants/Organs/SlimeCore.cs

```diff
@@ -0,0 +1,222 @@
+	[NaughtyAttributes.Button()]
+
+	public void InitialiseBabySlime()
+	{
+		var Multiplier = (StartingAmount / ReagentCirculatedComponent.AssociatedSystem.BloodPool.Total);
+		ReagentCirculatedComponent.AssociatedSystem.BloodPool.Multiply(Multiplier);
+
+		foreach (var bodyPart in RelatedPart.HealthMaster.BodyPartList)
+		{
+			var Body =  bodyPart.GetComponent<SlimeBody>();
+
+			if (Body != null)
+			{
+				Body.BabySprite = BabySprite;
+				Body.AdultSprite = AdultSprite;
+				Body.CurrentState = null;
+			}
+
+		}
+	}
+
+	public override void OnAddedToBody(LivingHealthMasterBase livingHealth)
+	{
+
+		foreach (var bodyPart in livingHealth.BodyPartList)
+		{
+
+			var Body =  bodyPart.GetComponent<SlimeBody>();
+
+			if (Body != null)
+			{
+				Body.BabySprite = BabySprite;
+				Body.AdultSprite = AdultSprite;
+				Body.CurrentState = null;
+			}
+
+		}
+	} //Warning only add body parts do not remove body parts in this
+
+	[NaughtyAttributes.Button()]
+	public void DEBUGSlimesSplit()
+	{
+		var Multiplier =  (100f / ReagentCirculatedComponent.AssociatedSystem.BloodPool.Total);
+		ReagentCirculatedComponent.AssociatedSystem.BloodPool.Multiply(Multiplier);
+		SlimesSplit();
+	}
+
+	public bool CanSlimesSplit()
+	{
+		return ReagentCirculatedComponent.AssociatedSystem.BloodPool[SlimeJelly] > 99;
+	}
+
+	public void SlimesSplit()
+	{
+
+		if (CanSlimesSplit())
+		{
+
+			// Loop four times to create four offspring
+			for (int i = 0; i < 4; i++)
+			{
+				// Generate a random number between 0 and 100
+				int randomNumber = Random.Range(0, 100);
+
+				// Find the index of the chosen prefab based on the percentage chances
+				int chosenIndex = GetChosenIndex(randomNumber);
+
+				SlimeCore NewCore = CanSplitInto[chosenIndex].CoreMutateTo;
+
+				if (CanSplitInto[chosenIndex].CoreMutateTo == null)
+				{
+					//is it Self
+					NewCore = this;
+				}
+
+				// Instantiate a new offspring based on the chosen prefab
+				var Mind = PlayerSpawn.NewSpawnCharacterV2(null, NewCore.ForCore, true);
+				Mind.Body.ObjectPhysics.AppearAtWorldPositionServer(RelatedPart.HealthMaster.gameObject.AssumedWorldPosServer());
+
+				var core = Mind.Body.GetComponent<LivingHealthMasterBase>().brain.GetComponent<SlimeCore>();
+				core.InitialiseBabySlime();
+
+				var GoodPlayers = core.GetComponent<BrainSlime>();
+				GoodPlayers.GoodPlayers = GoodPlayers.GoodPlayers.Union(this.GetComponent<BrainSlime>().GoodPlayers).ToDictionary(s => s.Key, s => s.Value);
+
+				core.Stabilised = this.Stabilised;
+
+			}
+
+
+			_ = Despawn.ServerSingle(RelatedPart.HealthMaster.gameObject);
+		}
+
+
+	}
+
+	// Helper method to get the index of the chosen prefab based on the percentage chances
+    private int GetChosenIndex(int randomNumber)
+    {
+        int totalChances = 0;
+
+        // Calculate the total chances
+        for (int i = 0; i < CanSplitInto.Count; i++)
+        {
+	        if (CanSplitInto[i].CoreMutateTo == null)
+	        {
+		        if (this.Stabilised)
```

Comment:
why are you using `this` a lot? this isn't GML or javascript

Reply:
seems fine to be implicit